modules/core/src/main/scala/iota/internal/CopKFunctionKMacros.scala

```diff
@@ -51,7 +51,9 @@ final class CopKFunctionKMacros(val c: Context) {
       lookup = unorderedPairs.toMap
 
       arrs <- Traverse[List].traverse(tpes)(tpe =>
-        lookup.get(tpe).toRight(s"Missing interpreter $NatTransName[$tpe, $G]").toAvowalNel).toEither
+        lookup.get(tpe).orElse {
```

Comment:
I wonder if it's worth keeping `lookup` as a list of entries and just always using the `.find` call you're doing below. Have you tried this to see if the tests still pass?